sdcm/cluster.py

```diff
@@ -4447,35 +4447,26 @@ def wait_for_schema_agreement(self):
         self.log.debug('Schema agreement is reached')
         return True
 
-    def check_nodes_up_and_normal(self, nodes=None, verification_node=None):
-        """Checks via nodetool that node joined the cluster and reached 'UN' state"""
+    def check_nodes_up_and_normal(self, nodes: Optional[list[BaseNode]] = None, verification_node: Optional[BaseNode] = None):
+        """Checks via nodetool executed on verification node that nodes joined the cluster and reached 'UN' state"""
         if not nodes:
             nodes = self.nodes
         status = self.get_nodetool_status(verification_node=verification_node)
-        up_statuses = []
+        ip_statuses = {key: value for dc_value in status.values() for key, value in dc_value.items()}
         for node in nodes:
-            found_node_status = False
-            for dc_status in status.values():
-                ip_status = dc_status.get(node.ip_address)
-                if ip_status:
-                    found_node_status = True
-                    up_statuses.append(ip_status["state"] == "UN")
-                    break
-            if not found_node_status:
-                up_statuses.append(False)
-        if not all(up_statuses):
-            raise ClusterNodesNotReady("Not all nodes joined the cluster")
+            ip_status = ip_statuses.get(node.ip_address)
+            if not ip_status or ip_status["state"] != "UN":
+                raise ClusterNodesNotReady("Node %s with ip %s didn't join to the cluster", node.name, node.ip_address)
```

Comment:
It will be great to check all nodes and return lists of nodes that not already.

Reply:
added.